specification/storagetasks/resource-manager/Microsoft.StorageTasks/stable/2023-01-01/storagetasks.json

```diff
@@ -0,0 +1,1085 @@
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageTasks/storageTasks/{storageTaskName}/storageTaskAssignments": {
+      "get": {
+        "tags": [
+          "StorageTasks"
+        ],
+        "operationId": "ListStorageTaskAssignmentIds",
+        "description": "Lists all the storage tasks available under the given resource group.",
+        "x-ms-examples": {
+          "ListStorageTasksByResourceGroup": {
+            "$ref": "./examples/storageTasksList/ListStorageTaskAssignmentIds.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/StorageTaskName"
+          },
+          {
+            "name": "$maxpagesize",
+            "in": "query",
+            "type": "string",
+            "description": "Optional, specifies the maximum number of storage task assignment Ids to be included in the list response."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- List of storage task assignment Ids retrieved and returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/StorageTaskAssignmentsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageTasks/storageTasks/{storageTaskName}/reports": {
+      "get": {
+        "tags": [
+          "StorageTasks"
+        ],
+        "operationId": "ListStorageTasksRunReportSummary",
+        "description": "Fetch the storage tasks run report summary for each assignment.",
+        "x-ms-examples": {
+          "ListStorageTasksByResourceGroup": {
+            "$ref": "./examples/storageTasksList/ListStorageTasksRunReportSummary.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/StorageTaskName"
+          },
+          {
+            "name": "$maxpagesize",
```

Comment:
>$maxpagesize [](http://example.com/codeflow?start=21&length=12)

Is $maxpagesize a valid OData Query parameter? I believe, $top and $skipToken are used to achieve this.

Reply:
$top and $maxpagesize can be used to set page size. we have been using this in Microsoft.Storage

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/storage/resource-manager/Microsoft.Storage/stable/2023-01-01/storage.json

Reply:
However, OData specs don't shoe $maxpagesize: [https://www.odata.org/getting-started/basic-tutorial](https://www.odata.org/getting-started/basic-tutorial). But approving since being used as before.